File: test/leetcode/sliding_window/length-of-longest-substring.test.ts
import { lengthOfLongestSubstring } from '../../src/two-pointer/longest-substring-without-repeating-chars';

describe('length of longest substring without repeated characters', () => {
  test('length of longest substring - test case 1', async () => {
    expect(lengthOfLongestSubstring('abcabcbb')).toBe(3);
  });

  test('length of longest substring - test case 2', async () => {
    expect(lengthOfLongestSubstring('bbbbb')).toBe(1);
  });

  test('length of longest substring - test case 3', async () => {
    expect(lengthOfLongestSubstring('pwwkew')).toBe(3);
  });
});

File: test/leetcode/sliding_window/length_of_longest_substring_test.py
from leetcode.sliding_window.length_of_longest_substring import Solution


soln = Solution()


def test_case_1():
    assert soln.lengthOfLongestSubstring("abcabcbb") == 3


def test_case_2():
    assert soln.lengthOfLongestSubstring("bbbbb") == 1


def test_case_3():
    assert soln.lengthOfLongestSubstring("pwwkew") == 3
